### ISSUE TITLE
I have a question for AddComponent

### ISSUE THREAD

**Issue by yunli97** [ByNameModding/BNM-Android#63]
I created a new GameObject and tried to call AddComponent[new_GameObject].Call(MyClass::BNMCustomClass.type);. The following code can run correctly:
![image](https://github.com/user-attachments/assets/fdb1ad1c-498b-4cc4-abe4-798ccbdd3310)
However, when I used a thread to load the initialization without changing anything else, the program crashed when executing AddComponent.
![image](https://github.com/user-attachments/assets/d89fceba-3b5c-4801-8803-a5044f285f8d)
There were no crash logs.

What is this causing AddComponent to crash?

**Comment by yunli97** [comment]
I changed sleep(1) to usleep(1) and it works again.

**Comment by BNM-Dev** [comment]
Calling unity methods outside of main thread is dangerous. This is reason of crash. And you can't call it using AddOnLoadEvent too, this event is too early. Hook sth, like `Internal_SceneLoaded` and create object here.

**Comment by yunli97** [comment]
yes, I hook **AdsManager**(MonoBehaviour) .**ctor** method in OnLoaded_SomeMethod，and create object here.
![image](https://github.com/user-attachments/assets/071f0dbc-2bc2-4b2a-8dfd-1f8b944eb019)
Does this still count as calling a unity method outside of the main thread?

**Comment by yunli97** [comment]
I tried to use **AttachIl2Cpp()**，but this still crash ,when Code executed to **AddComponent[new_GameObject].Call(MyClass::BNMCustomClass.type);**

![image](https://github.com/user-attachments/assets/66d747f2-4872-4b91-a6cd-006bce7c7fbc)
![image](https://github.com/user-attachments/assets/7ac84c04-353c-4074-8933-8f432a499d2b)

From the logs, it seems that he is on the main thread

**Comment by BNM-Dev** [comment]
Try use BNM v2.1.0

**Comment by yunli97** [comment]
Running it in a sub-thread causes all sorts of weird problems, and I've given up on it

**Comment by doviettung96** [comment]
I confirmed that adding attach  and detach make it worked. Before my code crashed as well. Now it worked.

**Comment by doviettung96** [comment]
however, that works for most of my function but still crash on some Closure calling
